Vuetify published histories

itisAliRH opened the pull request:
Followup #14907 and #14893
This pr vuetified the `Published Histories` component.

![image](https://user-images.githubusercontent.com/8046843/203330780-260eb999-5c38-42d7-892a-b5c264e52cf6.png)

The current API has some limitations:
- filtering by `annotation` is case sensitive and does not return the right result (client always sends queries in lowercase)
- filtering by `owner/username` is not available
- sort by `owner/username` is not available

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Share a history
  2. Go to `/histories/list_published`

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

itisAliRH committed:
Add history servieces
`getPublishedHistories`

itisAliRH committed:
New published histories

itisAliRH committed:
Register new published histories component

guerler commented on client/src/components/History/services.js:
```diff
@@ -0,0 +1,32 @@
+import { rethrowSimple } from "utils/simple-error";
+
+export class Services {
+    async getPublishedHistories(options = {}) {
```
This can be exported directly as a function without the `Services` class wrapper. Since the class is stateless.

itisAliRH committed:
History services exported directly, add `updateTags` function

itisAliRH committed:
History published: update services usage, create a new tag directly in the table, add button titles

itisAliRH committed:
history filtering: add `annotation` filter in valid filters, add `useDefaultFilters` for `getFilters` and `getQyeryDict`

itisAliRH committed:
history services: use `getQueryDict` to parse filter text into url query

itisAliRH committed:
HistoryPublished: aad `onTagClick` to search with tag, add ids for tests

itisAliRH committed:
add `published_histories` selenium navigations

itisAliRH committed:
delete `test_published_histories_grid` selenium test

itisAliRH committed:
add selenium tests for published histories: `test_histories_published`

itisAliRH committed:
replace published history test name in `mypy.ini`
`test_published_histories_grid` to `test_histories_published`

itisAliRH committed:
delete `HistoryAllPublishedGrid` class and related lines from history controller

itisAliRH committed:
delete unused imports from history controller

guerler commented on client/src/components/History/HistoryPublishedList.vue:
```diff
@@ -0,0 +1,233 @@
+import DebouncedInput from "components/DebouncedInput";
+import { getPublishedHistories, updateTags } from "./services";
+import { getFilters, toAlias } from "store/historyStore/model/filtering";
+import { getFilterText } from "./CurrentHistory/HistoryFilters/filterConversion";
```
We should move this file to a shared location maybe `utils/filter.js`.

guerler commented on client/src/components/History/HistoryPublishedList.vue:
```diff
@@ -0,0 +1,233 @@
+import LoadingSpan from "components/LoadingSpan";
+import DebouncedInput from "components/DebouncedInput";
+import { getPublishedHistories, updateTags } from "./services";
+import { getFilters, toAlias } from "store/historyStore/model/filtering";
```
We should refactor this so that other published components can use these `filterOperations` without having to import history store internals. I think we should move parts of this functionality to a shared directory and consolidate it in `utils/filter.js`. `validFilters` is probably the only function which should remain in the history context, since it is history specific.

guerler commented on client/src/components/History/services.js:
```diff
@@ -0,0 +1,41 @@
+import { rethrowSimple } from "utils/simple-error";
+import { getQueryDict } from "store/historyStore/model/filtering";
+
+const getQueryString = (filterText) => {
```
This should also be moved to `filter.js`, its likely a duplicate and is already used in `components/History/model/queries.js` and `store/historyStore/historyItemsStore.js`.

itisAliRH committed:
move `filterConversion.js` from `HistoryFilters` to `utils` to accessible for other components

itisAliRH committed:
update component imports using `filterConversation`

itisAliRH committed:
move all filtering methods/vars from `historyStore/model/filtering` to `utils/filterConversation`

itisAliRH committed:
update all imports in components/services/tests using filtering methods
`components/History/Content/SelectedItems.js`
`components/History/CurrentHistory/HistoryFilters/HistoryFilters.test.js`
`components/History/CurrentHistory/HistoryFilters/HistoryFilters.vue`
`components/History/CurrentHistory/HistoryOperations/SelectionOperations.vue`
`components/History/HistoryPublishedList.vue`
`components/History/services.js`
`store/historyStore/historyItemsStore.js`
`utils/filterConversion.test.js`
`utils/filtering.test.js`

itisAliRH committed:
set `useDefaultFilters` default for `getQueryString` to `true`

itisAliRH committed:
fix selenium error

get all published histories from server to make sure all of them are included in assertions

itisAliRH committed:
add `get_published_history_names_from_server` to histories published test

guerler commented on client/src/utils/filterConversion.js:
```diff
@@ -10,41 +10,53 @@
 
 /** Converts user input for case-insensitive filtering. */
-function toLower(value) {
+export const toLower = (value) => {
```
Talked about this, we intend to restore the `function NAME() {}` notation here. More details will follow separately. We intend to add this and other conventions to the docs for clarificiation but thats not part of this PR.

itisAliRH committed:
restore  notation

guerler reviewed (approved):
Looks good to me. Thank you so much.
